bindings/java/src/main/com/apple/foundationdb/EventKeeper.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * EventKeeper.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apple.foundationdb;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A device for externally instrumenting the FDB java driver, for monitoring
+ * purposes.
+ *
+ * Note that implementations as expected to be thread-safe, and may be manipulated
+ * from multiple threads even in nicely single-threaded-looking applications.
+ */
+public interface EventKeeper {
```

Comment:
The intent has always been to integrate this in with `StoreTimer`; my assumption has been that `StoreTimer` would implement this interface, and the implementation would translate the FDB-specific events into RecordLayer constructs for counting, rather than directly using the Events objects here. Having not written that part yet, I'm not sure if that is exactly what will end up happening, but I imagine something close to that.

Reply:
I think the way that this is written, I'm not certain that `StoreTimer.Event`s can be `EventKeeper.Event`s, so I think you'd need the `StoreTimer` to have-a shim `EventKeeper` that it created/passed along to the FDB level. Which isn't so bad, but it is a bit of machinery that I'm not sure we'd have done if we'd started the other way ¯\\\_(ツ)_/¯

I guess this is fine, as long as we're willing to make the shim as well. I'm not sure if there's a great alternative, really, other than making the APIs identical and/or making the `EventKeeper` API a subset of the `StoreTimer` API (and same for the events) and giving them identical semantics.

Reply:
it is not my expectation that `StoreTimer.Event` would be an `EventKeeper.Event`. Instead, it's more likely to be something like
```
private class FDBStoreTimer implements EventKeeper{
    private Map<EventKeeper.Event,StoreTimer.Event> translationMap;
    count(EventKeeper.Event event, long amt){
        StoreTimer.Event sEvent translationMap.get(event);
        if(sEvent == null){
             //we don't care about this event
        }else {
            StoreTimer.this.record(sEvent,amt); //or however the StoreTimer interface is constructed
        }
    }
}
```

which should accomplish the need to integrate the two. 

Since the two projects are entirely separate, the only way of avoiding a translation layer would be to move `StoreTimer` entirely into FDB proper, and (from my rather short perusal of that code) `StoreTimer` has a lot of record-layer specific stuff that would be painful to deal with. Having a translation layer like that above is a cheap and easy way of avoiding the refactor-across-projects problem